distributed/utils.py

```diff
@@ -1390,7 +1390,6 @@ def is_valid_xml(text):
 
 
 _offload_executor = ThreadPoolExecutor(max_workers=1, thread_name_prefix="Dask-Offload")
-weakref.finalize(_offload_executor, _offload_executor.shutdown)
```

Comment:
This is just an attempt to fix the error I'm seeing

Two reasons why I believe this should be removed regardless of whether this is a fix or not
1. All python versions 3.8+ are ensuring that worker threads are terminating on interpreter shutdown already. They explicitly handle the case of collected executors, interpreter shutdown and instance shutdown identically.
2. Judging by the [`finalize`](https://docs.python.org/3/library/weakref.html#weakref.finalize) docs I'm not even sure if this callback is ever triggered

> A finalizer will never invoke its callback during the later part of the [interpreter shutdown](https://docs.python.org/3/glossary.html#term-interpreter-shutdown) when module globals are liable to have been replaced by [None](https://docs.python.org/3/library/constants.html#None).

since `_offload_executor` is a module global...

Reply:
this is not responsible after all but I still suggest to remove this line

Reply:
yes, that note is interesting as well, i.e. this finalize is useless for many reasons

Reply:
I opened https://github.com/dask/distributed/issues/7639

Our code base is riddled with this pattern

Reply:
This one line makes me *very* nervous. Everything you wrote makes perfect sense, but *just in case* you're wrong, could you move it to its own PR so that it's not ending up in the release? It has the potential of leaving workers stuck on shutdown, and it also has the potetial of different behaviour on different Python versions and on different OSs, so I believe some thorough testing is in order.